### ISSUE TITLE
DiagnosticAnalyzers should ignore static members of grain interfaces

### ISSUE THREAD

**Issue by minichma** [dotnet/orleans#8501]
As static members of grain interfaces are inaccessible via grain references, they should be ignored by the analyzers but as of today they are only ignored in a limited number of cases.

E.g.: The diagnostic analyzers incorrectly issue compile errors `ORLEANS0002`, `ORLEANS0008` and `ORLEANS0009` for the following interface:

```cs
    public interface IFoo : IGrainWithStringKey
    {
        // error ORLEANS0002: Reference parameter modifiers are not allowed
        public static string StaticMethodWithOutPar(out string x)
        {
            x = "";
            return "";
        }

        // error ORLEANS0008: Grain interfaces must not contain properties
        public static string StaticProperty => "x";

        // error ORLEANS0009: Grain interfaces methods must return a compatible type, such as Task, Task<T>, ValueTask, ValueTask<T>, or void
        public static virtual string VirtualOrAbstractStaticMethod(string x) => "";
    }
```

Static members have partly been addressed in #8389 (v7.1.2), but the mentioned issues still exist.

**Comment by ReubenBond** [comment]
Thank you for pointing this out, @minichma, if you are willing to open a PR, that would be much appreciated, otherwise we will try to get to it soon.

**Comment by minichma** [comment]
@ReubenBond Sure, see https://github.com/dotnet/orleans/pull/8506. Previously the checks have explicitly been kept for static abstract and static virtual methods (see https://github.com/dotnet/orleans/pull/8389/commits/2aaae262eb90d5931adad12af77b47e136b96425), but I think that's wrong, so I changed that as well. Or am I missing something? @cmeyertons?

**Comment by minichma** [comment]
Closing this as it has been fixed with #8506.
